File: 151051/Makovetskiy/task330+340+350/dc-rest/src/main/java/com/rmakovetskij/dc_rest/controller/LabelController.java
package com.rmakovetskij.dc_rest.controller;

import com.rmakovetskij.dc_rest.model.dto.requests.LabelRequestTo;
import com.rmakovetskij.dc_rest.model.dto.responses.LabelResponseTo;
import com.rmakovetskij.dc_rest.service.LabelService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/labels")
@RequiredArgsConstructor
public class LabelController {

    private final LabelService labelService;

    @PostMapping
    public ResponseEntity<LabelResponseTo> createLabel(@RequestBody @Valid LabelRequestTo labelRequestDto) {
        LabelResponseTo createdLabel = labelService.createLabel(labelRequestDto);
        return new ResponseEntity<>(createdLabel, HttpStatus.CREATED);
    }

    @GetMapping
    public ResponseEntity<List<LabelResponseTo>> getAllLabels() {
        List<LabelResponseTo> labels = labelService.getAllLabels();
        return new ResponseEntity<>(labels, HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<LabelResponseTo> getLabelById(@PathVariable Long id) {
        LabelResponseTo label = labelService.getLabelById(id);
        return new ResponseEntity<>(label, HttpStatus.OK);
    }

    @PutMapping("/{id}")
    public ResponseEntity<LabelResponseTo> updateLabel(@PathVariable Long id, @RequestBody @Valid LabelRequestTo labelRequestDto) {
        LabelResponseTo updatedLabel = labelService.updateLabel(id, labelRequestDto);
        return new ResponseEntity<>(updatedLabel, HttpStatus.OK);
    }

    @PutMapping
    public ResponseEntity<LabelResponseTo> updateStory(@RequestBody @Valid LabelRequestTo labelRequestDto) {
        if (labelRequestDto.getId() == null) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }

        LabelResponseTo updatedLabelResponseDto = labelService.updateLabel(labelRequestDto.getId(), labelRequestDto);
        return new ResponseEntity<>(updatedLabelResponseDto, HttpStatus.OK);
    }


    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteLabel(@PathVariable Long id) {
        try{
        labelService.deleteLabel(id);
        return new ResponseEntity<>(HttpStatus.NO_CONTENT); }
        catch (RuntimeException e) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
    }
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/controller/LabelController.kt
package bsuir.dc.rest.controller

import bsuir.dc.rest.dto.from.LabelFrom
import bsuir.dc.rest.dto.to.LabelTo
import bsuir.dc.rest.service.LabelService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1.0/labels")
class LabelController(
    private val labelService: LabelService
) {

    @PostMapping
    fun createLabel(@RequestBody @Valid labelFrom: LabelFrom): ResponseEntity<LabelTo> {
        val createdLabel = labelService.createLabel(labelFrom)
        return ResponseEntity(createdLabel, HttpStatus.CREATED)
    }

    @GetMapping("/{id}")
    fun getLabelById(@PathVariable id: Long): ResponseEntity<LabelTo> {
        val label = labelService.getLabelById(id)
        return ResponseEntity.ok(label)
    }

    @GetMapping
    fun getAllLabels(): ResponseEntity<List<LabelTo>> {
        val labels = labelService.getAllLabels()
        return ResponseEntity.ok(labels)
    }

    @PutMapping()
    fun updateLabel(@RequestBody @Valid labelFrom: LabelFrom): ResponseEntity<LabelTo> {
        val updatedLabel = labelService.updateLabel(labelFrom.id, labelFrom)
        return ResponseEntity.ok(updatedLabel)
    }

    @DeleteMapping("/{id}")
    fun deleteLabel(@PathVariable id: Long): ResponseEntity<Void> {
        labelService.deleteLabel(id)
        return ResponseEntity.noContent().build()
    }

    @GetMapping("/by-issue/{issueId}")
    fun getLabelsByIssueId(@PathVariable issueId: Long): ResponseEntity<List<LabelTo>> {
        val labels = labelService.getLabelsByIssueId(issueId)
        return ResponseEntity.ok(labels)
    }
}
